intrahost.py

```diff
@@ -293,16 +295,17 @@ def parse_alleles_string(allelesStr):
     return alleleCounts
 
 
-def get_mpileup_allele_counts(inBam, chrom, pos, inConsFasta):
+def get_mpileup_allele_counts(inBam, chrom, pos, inConsFasta, samtools=None):
     """ Return {allele : [forwardCount, reverseCount], ...}
         allele is:
             Iins for insertions where ins represents the inserted bases
             Dlen for deletions where len is the length of the deletion
             base itself for non-indels
             'i' or 'd', in which case report count for consensus.
     """
+    samtools = samtools or SamtoolsTool()
```

Comment:
My gut is that it's not so much the instantiation of the class that costs time, but rather the `install_and_get_path` invocation every time. I'm not sure if pre-creating this object and passing it around aggressively will necessarily result in the same time savings as perhaps just turning the Conda-only Tool classes into something that doesn't even bother checking/verifying proper installation before executing it.